### PR TITLE
feat: add HW5 for fall 2022

### PR DESCRIPTION
**Pull request by wodeni**
# Description

This year, the assignment will ask students to use analysis tools in Teedy. Therefore, some high-level changes are made:

* Allow students to select both JavaScript and Java analysis tools.
  * The pool of Java tools is not big, so we decided to add in JavaScript tools so students can test their angular frontend.
* Reduce the number of tools in the starter list for Java: since we are requiring students to find tools outside of the starter list, I reduced the number of Java tools in the list so they have more options.

## Minor changes

- [x] provide Java and JavaScript static and dynamic analysis tools starter list
- [x] update due dates
- [x] change template repo link
- [x] make sure points add up to `100`

## Open questions

* Students have never learned how to configure Maven plugins or JS packages. Should we provide any pointers?

### CODE REVIEW

**Comment by MichaelHilton** [comment]
added some inline comments above

**Review by MichaelHilton — COMMENTED** [review]
see inline changes

File: assignments/HW5.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+- [YourKit](https://www.yourkit.com/java/profiler/features/), commercial Java profiler (15 days trial)
+
+**JavaScript**
+
+Static analysis:
+- [eslint](https://eslint.org/) is a widely used linter in modern JavaScript packages. 
+- [flow](https://flow.org/) is a static type checker for JavaScript.
+
+Dynamic analysis:
+- [jalangi2](https://github.com/Samsung/jalangi2]) is a dynamic analysis framework for JavaScript.
```
Comment (rohanpadhye) [inline assignments/HW5.md:58]: Link is broken. I think there is an extra `]` in the URL.

**Comment by rohanpadhye** [comment]
Looks good overall! Nice idea including JavaScript tools. I think it might be hard for students to run dynamic tools on the JS part of the code, since there are no (to my knowledge) clean entry points to execute the JS in CI. Maybe only include static tools here?

**Comment by wodeni** [comment]
> Looks good overall! Nice idea including JavaScript tools. I think it might be hard for students to run dynamic tools on the JS part of the code, since there are no (to my knowledge) clean entry points to execute the JS in CI. Maybe only include static tools here?

Yeah that makes sense. The dynamic ones are all pretty academic and not very off-the-shelf. Removing them instead.

File: assignments/HW5.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+**Part B: ML Explainability -- Group (due Nov 18) -- 50 points (50%)**
+
+For this deliverable, you will be collecting data, and writing a report.  The report should include the data you collect as well as your interpretation of the data.
+
+First, you should collect data by running LIME on your ML model from the last homework (use the model you have in the `fall-22-hw4` repository), with the new data.  You should present the results of this as data in your report.
+
+Then, you will interpret the data to explain why your machine learning model is making predictions. This information should include the features that provide the most predictive power.
+
+You should also evaluate your machine learning model considering fairness issues. You will evaluate the performance of your model with a specific target fairness strategy in mind, and if you are unhappy with the fairness of the model, you will come up with thresholds that you feel the model must meet before you would feel comfortable using it.
```
[the remark this answers is not in the document]
Comment (wodeni) [reply assignments/HW5.md:114]: Added the spec in both place. If it's not obvious enough lmk!

File: assignments/HW5.md (hunk 1 above)
Comment (wodeni) [reply assignments/HW5.md:58]: Good catch. Thanks!

**Pull request merged**